src/segment_git.go

```diff
@@ -131,12 +146,38 @@ func (g *git) string() string {
 	if g.props.getBool(StatusColorsEnabled, false) {
 		g.SetStatusColor()
 	}
+	enableHyperlink := g.props.getBool(EnableHyperlink, false)
 	buffer := new(bytes.Buffer)
+
+	// url
+	upstream := replaceAllString("/.*", g.repo.upstream, "")
+	g.repo.url = g.getGitCommandOutput("remote", "get-url", upstream)
+
 	// branchName
 	if g.repo.upstream != "" && g.props.getBool(DisplayUpstreamIcon, false) {
-		fmt.Fprintf(buffer, "%s", g.getUpstreamSymbol())
+		fmt.Fprintf(buffer, "%s", g.getUpstreamSymbol(g.repo.url))
+	}
+	if g.repo.upstream != "" && enableHyperlink {
+		repoURL := strings.ReplaceAll(g.repo.url, ".git", "")
+		baseURL, _ := url.Parse(g.repo.url)
+		var urlTemplate string
+		// check for a match in the providers map
+		for i, val := range gitProvidersMap {
+			if strings.Contains(baseURL.Host, i) {
+				urlTemplate = val
+				break
+			}
+		}
+		if urlTemplate != "" {
+			// if provider found, open the branch
```

Comment:
```suggestion
			// if provider found, open the branch

			// normalize azure devops url to change from
			// https://{account}@dev.azure.com/{account}/{project}/_git/{repo}
			// to
			// https://dev.azure.com/{account}/{project}
			repoURL = replaceAllString("//[^@]*@", repoURL, "//")						
```

This suggestion makes the code work for me, for azure devops repositories.  I suspect you'll also have problems with people who have cloned using ssh, instead of https. I found a couple other repos which are doing something similar to what you're doing, which may be useful to review.

[dotnet/docfx](https://github.com/dotnet/docfx/blob/dev/src/docfx.website.themes/common/common.js) is probably the easiest to follow
[eamodio/vscode-gitlens](https://github.com/eamodio/vscode-gitlens/tree/main/src/git/remotes) has this functionality too

Hope this helps!

Reply:
Thanks. I thought the host property was stripping the account and returning the hostname only. I’ll check this.